File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length-1;
        int currmax = 0, ans = 0;
        while(right > left){
            int width = right - left;
            int hgt = Math.min(height[left], height[right]);
            currmax = width * hgt;
            ans = Math.max(currmax, ans);
            if (height[left] > height[right]){
                right--;
            }
            else{
                left++;
            }
        }
        return ans;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, currmax, ans = 0, 0, 0
        right = len(height)-1
        while(left < right):
            width = right - left
            currmax = (min(height[right], height[left])) * width
            ans = max(currmax, ans)
            if (height[right] > height[left]):
                left += 1
            else:
                right -= 1
        return ans
            
        